game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/AirBattleOffenseCombatValue.java

```diff
@@ -0,0 +1,93 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.TerritoryEffect;
+import games.strategy.engine.data.Unit;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Value;
+import org.triplea.java.collections.IntegerMap;
+
+/** Calculates offense strength and roll for Air Battle dice */
+@Builder
+@Value
+@Getter(AccessLevel.NONE)
+class AirBattleOffenseCombatValue implements CombatValue {
+
+  @Getter(onMethod = @__({@Override}))
+  @NonNull
+  GameData gameData;
+
+  @Getter(onMethod = @__({@Override}))
+  @NonNull
+  @Builder.Default
+  Collection<Unit> friendUnits = List.of();
+
+  @Getter(onMethod = @__({@Override}))
+  @NonNull
+  @Builder.Default
+  Collection<Unit> enemyUnits = List.of();
+
+  @Getter(onMethod = @__({@Override}))
+  @NonNull
+  @Builder.Default
+  Collection<TerritoryEffect> territoryEffects = List.of();
+
+  @Override
+  public RollCalculator getRoll() {
+    return new AirBattleOffenseRoll();
+  }
+
+  @Override
+  public StrengthCalculator getStrength() {
+    return new AirBattleOffenseStrength(gameData.getDiceSides());
+  }
+
+  @Override
+  public boolean isDefending() {
+    return false;
+  }
+
+  @Override
+  public int getDiceSides(final Unit unit) {
```

Comment:
It's really odd that `getDiceSides` has `Unit` as a parameter. AFAIK we do not support per-unit dice sides, it's a global config for a game. Why is the `unit` parameter here? Do we ever user `Unit` to compute the dice sides value?

Reply:
AA units have per-unit type dice sides.  Look at `AaDefenseCombatValue#getDiceSides` and `AaOffenseCombatValue#getDiceSides`